flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -715,15 +715,8 @@ abstract class TableEnvironment(val config: TableConfig) {
       functionName: String):
     GeneratedFunction[MapFunction[Row, OUT], OUT] = {
 
-    // validate that at least the field types of physical and logical type match
-    // we do that here to make sure that plan translation was correct
-    if (schema.physicalTypeInfo != inputTypeInfo) {
-      throw TableException("The field types of physical and logical row types do not match." +
-        "This is a bug and should not happen. Please file an issue.")
-    }
-
-    val fieldTypes = schema.physicalFieldTypeInfo
-    val fieldNames = schema.physicalFieldNames
+    val fieldTypes = schema.logicalFieldTypeInfo
+    val fieldNames = schema.logicalFieldNames
```

Comment:
I'm not sure about this change. The physical and logical type check is to make sure all the time indicators are translated.